akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.stream.io.Implicits._
+import akka.stream.scaladsl._
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import scala.util.Try
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Collects the bytes of the first file part from the request into the given destination file
+   *
+   * @param destination Will be overwritten if not empty
+   */
+  def uploadedFile(destination: File): Directive1[Try[FormFile]] =
```

Comment:
My suggestion was to keep both, with the simple one being based on the flexible one. ;)